### PR TITLE
Fix nxos_system tests

### PR DESCRIPTION
**Pull request by mikewiebe**
##### SUMMARY
* Moves `targets/nxos_system/tests/cli/sanity.yaml` inside a block so that if a step fails the username is reconfigured.  WIthout this other test can fail that require a specific username.
* Adds an ignore_errors: yes check for a teardown step.  Nxapi will error out when the domain does not exist.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
nxos_system

##### ANSIBLE VERSION
```
ansible 2.6.0 (rel250/fix_nxos_user 5a98076eb3) last updated 2018/02/14 13:01:12 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/Users/mwiebe/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mwiebe/Projects/nxos_ansible/fix_ansible/lib/ansible
  executable location = /Users/mwiebe/Projects/nxos_ansible/fix_ansible/bin/ansible
  python version = 2.7.13 (default, Apr  4 2017, 08:47:57) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]
```

### CODE REVIEW

**Comment by mikewiebe** [comment]
@trishnaguha  nxos_system test fixes.

**Comment by trishnaguha** [comment]
@mikewiebe Can we move the sanity test in common directory because I don't find any difference between cli and nxapi test?

File: test/integration/targets/nxos_system/tests/cli/sanity.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+- debug: msg="START TRANSPORT:CLI nxos_system sanity test"
```
Comment (trishnaguha) [inline test/integration/targets/nxos_system/tests/cli/sanity.yaml]: Please update the debug statement to so we know which connection plugin we are running:
```
- debug: msg="START connection={{ ansible_connection }}/sanity.yaml"
- debug: msg="Using provider={{ connection.transport }}"
  when: ansible_connection == "local"
```

File: test/integration/targets/nxos_system/tests/cli/sanity.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      name_servers:
+        - { server: 8.8.8.8, vrf: management }
+        - { server: 8.8.4.4, vrf: management }
+      provider: "{{ cli }}"
+
+  always:
+  - name: Re-configure hostname
+    nxos_system: *hostname
+
+  - debug: msg="END TRANSPORT:CLI nxos_system sanity test"
```
Comment (trishnaguha) [inline test/integration/targets/nxos_system/tests/cli/sanity.yaml]: Please update the debug statement to
`- debug: msg="END connection={{ ansible_connection }}/sanity.yaml"`

**Review by trishnaguha — CHANGES_REQUESTED** [review]

**Review by trishnaguha — APPROVED** [review]

**Comment by trishnaguha** [comment]
cherry-picked to stable-2.5

**Pull request merged**